op-challenger/game/fault/types/game.go

```diff
@@ -167,7 +170,8 @@ func (g *gameState) getParent(claim Claim) *extendedClaim {
 
 func makeClaimEntry(claim Claim) claimEntry {
 	return claimEntry{
-		ClaimData:           claim.ClaimData,
+		GIndex:              claim.ToGIndex().Uint64(),
```

Comment:
Using the `ClaimData` object itself has a `*big.Int` pointer internally so comparisons between the objects become flaky since the internals of the `big.Int` can be nil if not instantiated properly

Reply:
hmm, but we can't use a uint64 here as GIndex will need to exceed its max value. I'd be concerned if we're not instantiating objects properly - that's going to lead to all kinds of trouble regardless of whether you get a `nil` or a `0` since 0 is an invalid GIndex anyway.

Reply:
Ok I was having issues with the big ints in tests cases that compare the objects - it's that pesky big int internal NaN representation vs nil

Reply:
So, dug into this a little bit more.

The main change here was that we now use a `*big.Int` pointer inside the `Position` struct for the `IndexAtDepth` value.

Then the `claimEntry` embeds a `ClaimData` internally which itself has a `Position` embedded. When using the `claimEntry` as a key in the `claims` mapping, the struct is _comparable_ because the big int pointer is comparable inside the position. It is not if the big int is not a pointer.

So the reason why using `ClaimData` fails with this big int pointer, is that checking duplicate claims fails since a a new `claimEntry` will be constructed for the duplicate claim and can hold a different big int pointer internally. Thus the `IsDuplicate` check fails.

I've added a game test to demonstrate this using a deep clone of the `ClaimData`.

Since we cannot use a `big.Int` as it is not comparable and does not work as a key in the map, I think the best way around this is to build a canonical claim identifier for the claim state. I have done a naive version of this in this pr by hashing the claim entry values. It's not the most efficient but it works.

Would appreciate some feedback on how you want to proceed with this. It's not clear to me what approach we should take to get this pr over the line.

Reply:
hmm, yeah it appears that we can't avoid moving to something like a Hash - I don't like it too much but it is what it is.  My suggestion then would be to simplify the heck out of the game implementation like https://github.com/ethereum-optimism/optimism/pull/7517 - I _think_ with that, we'll just need to `computeClaimID` it adds to handle `ToGIndex` returning a big.Int instead of having converting it to big.Int. This is inline with what I was suggesting in terms of calculating claim ID but I realised it was a good opportunity to simplify the game implementation a lot and would work as a separate PR.

Reply:
Thank you so much for doing this - really meaningfully made the implementation cleaner